lib/carbon/cache.py

```diff
@@ -112,23 +112,25 @@ def _generate_queue():
 
     self.queue = _generate_queue()
 
-  def choose_item(self):
+  def choose_item(self, mdpu=False):
     return next(self.queue)
 
 
 class TimeSortedStrategy(DrainStrategy):
-  """ This strategy prefers metrics wich are lagging behind
+  """ This strategy prefers metrics which are lagging behind
   guarantees every point gets written exactly once during
   a loop of the cache """
   def __init__(self, cache):
     super(TimeSortedStrategy, self).__init__(cache)
+    self.mdpu_flag = False
 
     def _generate_queue():
       while True:
         t = time.time()
         metric_lw = sorted(self.cache.watermarks, key=lambda x: x[1], reverse=True)
         if settings.MIN_TIMESTAMP_LAG:
-          metric_lw = [x for x in metric_lw if t - x[1] > settings.MIN_TIMESTAMP_LAG]
+          metric_lw = [x for x in metric_lw if t - x[1] > settings.MIN_TIMESTAMP_LAG
+                        or (self.mdpu_flag and x[1] >= settings.MIN_DATAPOINTS_PER_UPDATE >0)]
```

Comment:
Can you explain what's going on here?

Reply:
Riiight, I probably screwed up things here. I should use cache.counts() instead of watermarks.  Back to drawing board!